vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/pipelines/inline_content_ads/eligible_inline_content_ads_v1.cc

```diff
@@ -97,41 +109,53 @@ void EligibleAdsV1::GetForChildSegments(
   const database::table::CreativeInlineContentAds database_table;
   database_table.GetForSegmentsAndDimensions(
       segments, dimensions,
-      [=](const bool success, const SegmentList& /*segments*/,
-          const CreativeInlineContentAdList& creative_ads) {
-        if (!success) {
-          BLOG(1, "Failed to get ads for child segments");
-          callback(/*had_opportunity*/ false, {});
-          return;
-        }
-
-        const CreativeInlineContentAdList eligible_creative_ads =
-            FilterCreativeAds(creative_ads, ad_events, browsing_history);
-        if (eligible_creative_ads.empty()) {
-          BLOG(1, "No eligible ads out of " << creative_ads.size()
-                                            << " ads for child segments");
-          GetForParentSegments(user_model, dimensions, ad_events,
-                               browsing_history, callback);
-          return;
-        }
-
-        BLOG(1, eligible_creative_ads.size()
-                    << " eligible ads out of " << creative_ads.size()
-                    << " ads for child segments");
-
-        callback(/*had_opportunity*/ true, eligible_creative_ads);
-      });
+      base::BindOnce(&EligibleAdsV1::OnGetForChildSegments,
+                     base::Unretained(this), std::move(user_model), dimensions,
+                     ad_events, browsing_history, std::move(callback)));
+}
+
+void EligibleAdsV1::OnGetForChildSegments(
+    targeting::UserModelInfo user_model,
```

Comment:
clang: The parameter 'user_model' is copied for each invocation but only used as a const reference; consider making it a const reference

Reply:
fixed